File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    let stack = []
    const burger = [1,2,3,1]
    ingredient.forEach(v => {
        stack.push(v)
        if (burger.every((v,i) => stack.at(-4+i) === v)) {
            burger.forEach(() => stack.pop())
        }
    })
    return (ingredient.length-stack.length)/4
}
// function solution(ingredient) {
    
// }

File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    res = 0
    stack = []
    for i in ingredient:
        if i == 1 and len(stack) >= 3 and stack[-3:] == [1,2,3]:
            for _ in range(3): stack.pop()
            res += 1
        else:
            stack.append(i)
    return res
            